From 3369f939e61c70c3a3e5a35724f253e399bb19ae Mon Sep 17 00:00:00 2001
From: "Ho, Yu Xin" <yu.xin.ho@intel.com>
Date: Fri, 14 Sep 2018 19:10:26 +0800
Subject: [PATCH 11/98] ASoC: Intel: Boards: Clean up for AIC3107 Machine
 Driver

Clean up for AIC3107 machine driver especially on the
back end DAI link to improve clarity

Signed-off-by: Ho, Yu Xin <yu.xin.ho@intel.com>
---
 sound/soc/intel/boards/apli_lhcrb_aic3107M.c | 104 +++++++++++++--------------
 sound/soc/intel/boards/apli_lhcrb_aic3107S.c |  91 +++++++++++------------
 2 files changed, 96 insertions(+), 99 deletions(-)
 mode change 100644 => 100755 sound/soc/intel/boards/apli_lhcrb_aic3107M.c
 mode change 100644 => 100755 sound/soc/intel/boards/apli_lhcrb_aic3107S.c

diff --git a/sound/soc/intel/boards/apli_lhcrb_aic3107M.c b/sound/soc/intel/boards/apli_lhcrb_aic3107M.c
old mode 100644
new mode 100755
index a07883e..28cd9de
--- a/sound/soc/intel/boards/apli_lhcrb_aic3107M.c
+++ b/sound/soc/intel/boards/apli_lhcrb_aic3107M.c
@@ -25,7 +25,7 @@
 #include <sound/pcm_params.h>
 #include "../../codecs/tlv320aic3x.h"
 
-static int apli_lfcrb_aic3107M_startup(struct snd_pcm_substream *substream)
+static int apli_lhcrb_aic3107M_startup(struct snd_pcm_substream *substream)
 {
 	int ret;
 	static unsigned int rates[] = { 48000 };
@@ -61,17 +61,11 @@ static int apli_lfcrb_aic3107M_startup(struct snd_pcm_substream *substream)
 	return ret;
 }
 
-static struct snd_soc_ops apli_lfcrb_aic3107M_ops = {
-	.startup = apli_lfcrb_aic3107M_startup,
+static struct snd_soc_ops apli_lhcrb_aic3107M_ops = {
+	.startup = apli_lhcrb_aic3107M_startup,
 };
 
 static const struct snd_kcontrol_new apli_controls[] = {
-	SOC_DAPM_PIN_SWITCH("SSP1 Speaker"),
-	SOC_DAPM_PIN_SWITCH("SSP1 Mic"),
-	SOC_DAPM_PIN_SWITCH("SSP2 Speaker"),
-	SOC_DAPM_PIN_SWITCH("SSP2 Mic"),
-	SOC_DAPM_PIN_SWITCH("SSP4 Speaker"),
-	SOC_DAPM_PIN_SWITCH("SSP4 Mic"),
 	SOC_DAPM_PIN_SWITCH("Headphone Jack"),
 	SOC_DAPM_PIN_SWITCH("Mic Jack"),
 };
@@ -79,10 +73,10 @@ static int apli_lfcrb_aic3107M_startup(struct snd_pcm_substream *substream)
 static const struct snd_soc_dapm_widget apli_widgets[] = {
 	SND_SOC_DAPM_SPK("SSP1 Speaker", NULL),
 	SND_SOC_DAPM_MIC("SSP1 Mic", NULL),
-	SND_SOC_DAPM_SPK("SSP2 Speaker", NULL),
-	SND_SOC_DAPM_MIC("SSP2 Mic", NULL),
-	SND_SOC_DAPM_SPK("SSP4 Speaker", NULL),
-	SND_SOC_DAPM_MIC("SSP4 Mic", NULL),
+	SND_SOC_DAPM_SPK("SSP2 PbTestPin", NULL),
+	SND_SOC_DAPM_MIC("SSP2 CpTestPin", NULL),
+	SND_SOC_DAPM_SPK("SSP4 PbTestPin", NULL),
+	SND_SOC_DAPM_MIC("SSP4 CpTestPin", NULL),
 	SND_SOC_DAPM_HP("Headphone Jack", NULL),
 	SND_SOC_DAPM_MIC("Mic Jack", NULL),
 	SND_SOC_DAPM_LINE("Line In", NULL),
@@ -117,25 +111,25 @@ static int apli_lfcrb_aic3107M_startup(struct snd_pcm_substream *substream)
 
 	/* Codec BE connections */
 	/* SSP1 follows Hardware pin naming */
-	{"SSP1 Speaker", NULL, "ssp0 Tx"},
-	{"ssp0 Tx", NULL, "codec2_out"},
+	{"SSP1 Speaker", NULL, "ssp1 Tx"},
+	{"ssp1 Tx", NULL, "codec2_out"},
 
-	{"codec2_in", NULL, "ssp0 Rx"},
-	{"ssp0 Rx", NULL, "SSP1 Mic"},
+	{"codec2_in", NULL, "ssp1 Rx"},
+	{"ssp1 Rx", NULL, "SSP1 Mic"},
 
 	/* SSP2 follows Hardware pin naming */
-	{"SSP2 Speaker", NULL, "ssp1 Tx"},
-	{"ssp1 Tx", NULL, "codec0_out"},
+	{"SSP2 PbTestPin", NULL, "ssp2 Tx"},
+	{"ssp2 Tx", NULL, "codec0_out"},
 
-	{"codec0_in", NULL, "ssp1 Rx"},
-	{"ssp1 Rx", NULL, "SSP2 Mic"},
+	{"codec0_in", NULL, "ssp2 Rx"},
+	{"ssp2 Rx", NULL, "SSP2 CpTestPin"},
 
 	/* SSP4 follows Hardware pin naming */
-	{"SSP4 Speaker", NULL, "ssp3 Tx"},
-	{"ssp3 Tx", NULL, "codec1_out"},
+	{"SSP4 PbTestPin", NULL, "ssp4 Tx"},
+	{"ssp4 Tx", NULL, "codec1_out"},
 
-	{"codec1_in", NULL, "ssp3 Rx"},
-	{"ssp3 Rx", NULL, "SSP4 Mic"},
+	{"codec1_in", NULL, "ssp4 Rx"},
+	{"ssp4 Rx", NULL, "SSP4 CpTestPin"},
 };
 
 static int tlv320aic3107_init(struct snd_soc_pcm_runtime *rtd)
@@ -194,6 +188,8 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 	return 0;
 }
 
+static const char pname[] = "0000:00:0e.0";
+
 /* apli digital audio interface glue - connects codec <--> CPU */
 static struct snd_soc_dai_link apli_lhcrb_aic3107_dais[] = {
 	/* Front End DAI links */
@@ -201,7 +197,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.name = "SSP1 Playback Port",
 		.stream_name = "AIC3107 Playback",
 		.cpu_dai_name = "System Pin 4",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.nonatomic = 1,
 		.dynamic = 1,
 		.codec_name = "snd-soc-dummy",
@@ -209,7 +205,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.trigger = {SND_SOC_DPCM_TRIGGER_POST,
 				SND_SOC_DPCM_TRIGGER_POST},
 		.dpcm_playback = 1,
-		.ops = &apli_lfcrb_aic3107M_ops,
+		.ops = &apli_lhcrb_aic3107M_ops,
 	},
 	{
 		.name = "SSP1 Capture Port",
@@ -217,19 +213,19 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.cpu_dai_name = "System Pin 5",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.init = NULL,
 		.dpcm_capture = 1,
 		.ignore_suspend = 1,
 		.nonatomic = 1,
 		.dynamic = 1,
-		.ops = &apli_lfcrb_aic3107M_ops,
+		.ops = &apli_lhcrb_aic3107M_ops,
 	},
 	{
 		.name = "SSP2 Playback Port",
 		.stream_name = "SSP2 Speaker",
 		.cpu_dai_name = "System Pin",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.nonatomic = 1,
 		.dynamic = 1,
 		.codec_name = "snd-soc-dummy",
@@ -237,7 +233,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.trigger = {SND_SOC_DPCM_TRIGGER_POST,
 				SND_SOC_DPCM_TRIGGER_POST},
 		.dpcm_playback = 1,
-		.ops = &apli_lfcrb_aic3107M_ops,
+		.ops = &apli_lhcrb_aic3107M_ops,
 	},
 	{
 		.name = "SSP2 Capture Port",
@@ -245,19 +241,19 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.cpu_dai_name = "System Pin",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.init = NULL,
 		.dpcm_capture = 1,
 		.ignore_suspend = 1,
 		.nonatomic = 1,
 		.dynamic = 1,
-		.ops = &apli_lfcrb_aic3107M_ops,
+		.ops = &apli_lhcrb_aic3107M_ops,
 	},
 	{
 		.name = "SSP4 Playback Port",
 		.stream_name = "SSP4 Speaker",
 		.cpu_dai_name = "Deepbuffer Pin",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.nonatomic = 1,
 		.dynamic = 1,
 		.codec_name = "snd-soc-dummy",
@@ -265,7 +261,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.trigger = {SND_SOC_DPCM_TRIGGER_POST
 			, SND_SOC_DPCM_TRIGGER_POST},
 		.dpcm_playback = 1,
-		.ops = &apli_lfcrb_aic3107M_ops,
+		.ops = &apli_lhcrb_aic3107M_ops,
 	},
 	{
 		.name = "SSP4 Capture Port",
@@ -273,20 +269,20 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.cpu_dai_name = "Reference Pin",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.init = NULL,
 		.dpcm_capture = 1,
 		.ignore_suspend = 1,
 		.nonatomic = 1,
 		.dynamic = 1,
-		.ops = &apli_lfcrb_aic3107M_ops,
+		.ops = &apli_lhcrb_aic3107M_ops,
 	},
 	/* Back End DAI links */
 	{
-		/* SSP0 - Codec */
-		.name = "SSP0-Codec",
-		.be_id = 0,
-		.cpu_dai_name = "SSP0 Pin",
+		/* SSP1 - Codec */
+		.name = "SSP1-Codec",
+		.id = 0,
+		.cpu_dai_name = "SSP1 Pin",
 		.codec_name = "tlv320aic3x-codec.3-0018",
 		.codec_dai_name = "tlv320aic3x-hifi",
 		.ignore_suspend = 1,
@@ -295,7 +291,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.dai_fmt = SND_SOC_DAIFMT_I2S |
 					SND_SOC_DAIFMT_NB_NF |
 					SND_SOC_DAIFMT_CBS_CFS,
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.ignore_suspend = 1,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
@@ -303,13 +299,13 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.be_hw_params_fixup = aic3107_be_hw_params_fixup,
 	},
 	{
-		/* SSP1 - Codec */
-		.name = "SSP1-Codec",
-		.be_id = 1,
-		.cpu_dai_name = "SSP1 Pin",
+		/* SSP2 - Codec */
+		.name = "SSP2-Codec",
+		.id = 1,
+		.cpu_dai_name = "SSP2 Pin",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.ignore_suspend = 1,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
@@ -317,16 +313,16 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.init = NULL,
 	},
 	{
-		/* SSP3 - Codec */
-		.name = "SSP3-Codec",
-		.be_id = 2,
-		.cpu_dai_name = "SSP3 Pin",
+		/* SSP4 - Codec */
+		.name = "SSP4-Codec",
+		.id = 2,
+		.cpu_dai_name = "SSP4 Pin",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
 		.dai_fmt = SND_SOC_DAIFMT_I2S |
 					SND_SOC_DAIFMT_NB_NF |
 					SND_SOC_DAIFMT_CBS_CFS,
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.ignore_suspend = 1,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
@@ -394,7 +390,7 @@ static int apli_audio_remove(struct platform_device *pdev)
 	.probe = apli_audio_probe,
 	.remove = apli_audio_remove,
 	.driver = {
-		.name = "lfcrb_aic3107M_i2s",
+		.name = "lhcrb_aic3107M_i2s",
 	},
 };
 
@@ -403,4 +399,4 @@ static int apli_audio_remove(struct platform_device *pdev)
 /* Module information */
 MODULE_DESCRIPTION("Intel Audio aic3170 Machine driver  for APL-I LH CRB");
 MODULE_LICENSE("GPL v2");
-MODULE_ALIAS("platform:lfcrb_aic3107M_i2s");
+MODULE_ALIAS("platform:lhcrb_aic3107M_i2s");
diff --git a/sound/soc/intel/boards/apli_lhcrb_aic3107S.c b/sound/soc/intel/boards/apli_lhcrb_aic3107S.c
old mode 100644
new mode 100755
index 93e2ab9..d088717
--- a/sound/soc/intel/boards/apli_lhcrb_aic3107S.c
+++ b/sound/soc/intel/boards/apli_lhcrb_aic3107S.c
@@ -25,7 +25,7 @@
 #include <sound/pcm_params.h>
 #include "../../codecs/tlv320aic3x.h"
 
-static int apli_lfcrb_aic3107S_startup(struct snd_pcm_substream *substream)
+static int apli_lhcrb_aic3107S_startup(struct snd_pcm_substream *substream)
 {
 	int ret;
 	static unsigned int rates[] = { 48000 };
@@ -61,8 +61,8 @@ static int apli_lfcrb_aic3107S_startup(struct snd_pcm_substream *substream)
 	return ret;
 }
 
-static struct snd_soc_ops apli_lfcrb_aic3107S_ops = {
-	.startup = apli_lfcrb_aic3107S_startup,
+static struct snd_soc_ops apli_lhcrb_aic3107S_ops = {
+	.startup = apli_lhcrb_aic3107S_startup,
 };
 
 static const struct snd_kcontrol_new apli_controls[] = {
@@ -79,10 +79,10 @@ static int apli_lfcrb_aic3107S_startup(struct snd_pcm_substream *substream)
 static const struct snd_soc_dapm_widget apli_widgets[] = {
 	SND_SOC_DAPM_SPK("SSP1 Speaker", NULL),
 	SND_SOC_DAPM_MIC("SSP1 Mic", NULL),
-	SND_SOC_DAPM_SPK("SSP2 Speaker", NULL),
-	SND_SOC_DAPM_MIC("SSP2 Mic", NULL),
-	SND_SOC_DAPM_SPK("SSP4 Speaker", NULL),
-	SND_SOC_DAPM_MIC("SSP4 Mic", NULL),
+	SND_SOC_DAPM_SPK("SSP2 PbTestPin", NULL),
+	SND_SOC_DAPM_MIC("SSP2 CpTestPin", NULL),
+	SND_SOC_DAPM_SPK("SSP4 PbTestPin", NULL),
+	SND_SOC_DAPM_MIC("SSP4 CpTestPin", NULL),
 	SND_SOC_DAPM_HP("Headphone Jack", NULL),
 	SND_SOC_DAPM_MIC("Mic Jack", NULL),
 	SND_SOC_DAPM_LINE("Line In", NULL),
@@ -117,25 +117,25 @@ static int apli_lfcrb_aic3107S_startup(struct snd_pcm_substream *substream)
 
 	/* Codec BE connections */
 	/* SSP1 follows Hardware pin naming */
-	{"SSP1 Speaker", NULL, "ssp0 Tx"},
-	{"ssp0 Tx", NULL, "codec2_out"},
+	{"SSP1 Speaker", NULL, "ssp1 Tx"},
+	{"ssp1 Tx", NULL, "codec2_out"},
 
-	{"codec2_in", NULL, "ssp0 Rx"},
-	{"ssp0 Rx", NULL, "SSP1 Mic"},
+	{"codec2_in", NULL, "ssp1 Rx"},
+	{"ssp1 Rx", NULL, "SSP1 Mic"},
 
 	/* SSP2 follows Hardware pin naming */
-	{"SSP2 Speaker", NULL, "ssp1 Tx"},
-	{"ssp1 Tx", NULL, "codec0_out"},
+	{"SSP2 PbTestPin", NULL, "ssp2 Tx"},
+	{"ssp2 Tx", NULL, "codec0_out"},
 
-	{"codec0_in", NULL, "ssp1 Rx"},
-	{"ssp1 Rx", NULL, "SSP2 Mic"},
+	{"codec0_in", NULL, "ssp2 Rx"},
+	{"ssp2 Rx", NULL, "SSP2 CpTestPin"},
 
 	/* SSP4 follows Hardware pin naming */
-	{"SSP4 Speaker", NULL, "ssp3 Tx"},
-	{"ssp3 Tx", NULL, "codec1_out"},
+	{"SSP4 PbTestPin", NULL, "ssp4 Tx"},
+	{"ssp4 Tx", NULL, "codec1_out"},
 
-	{"codec1_in", NULL, "ssp3 Rx"},
-	{"ssp3 Rx", NULL, "SSP4 Mic"},
+	{"codec1_in", NULL, "ssp4 Rx"},
+	{"ssp4 Rx", NULL, "SSP4 CpTestPin"},
 };
 
 static int tlv320aic3107_init(struct snd_soc_pcm_runtime *rtd)
@@ -194,6 +194,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 	return 0;
 }
 
+static const char pname[] = "0000:00:0e.0";
 /* apli digital audio interface glue - connects codec <--> CPU */
 static struct snd_soc_dai_link apli_lhcrb_aic3107_dais[] = {
 	/* Front End DAI links */
@@ -201,7 +202,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.name = "SSP1 Playback Port",
 		.stream_name = "AIC3107 Playback",
 		.cpu_dai_name = "System Pin 4",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.nonatomic = 1,
 		.dynamic = 1,
 		.codec_name = "snd-soc-dummy",
@@ -209,7 +210,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.trigger = {SND_SOC_DPCM_TRIGGER_POST,
 				SND_SOC_DPCM_TRIGGER_POST},
 		.dpcm_playback = 1,
-		.ops = &apli_lfcrb_aic3107S_ops,
+		.ops = &apli_lhcrb_aic3107S_ops,
 	},
 	{
 		.name = "SSP1 Capture Port",
@@ -217,19 +218,19 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.cpu_dai_name = "System Pin 5",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.init = NULL,
 		.dpcm_capture = 1,
 		.ignore_suspend = 1,
 		.nonatomic = 1,
 		.dynamic = 1,
-		.ops = &apli_lfcrb_aic3107S_ops,
+		.ops = &apli_lhcrb_aic3107S_ops,
 	},
 	{
 		.name = "SSP2 Playback Port",
 		.stream_name = "SSP2 Speaker",
 		.cpu_dai_name = "System Pin",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.nonatomic = 1,
 		.dynamic = 1,
 		.codec_name = "snd-soc-dummy",
@@ -237,7 +238,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.trigger = {SND_SOC_DPCM_TRIGGER_POST,
 				SND_SOC_DPCM_TRIGGER_POST},
 		.dpcm_playback = 1,
-		.ops = &apli_lfcrb_aic3107S_ops,
+		.ops = &apli_lhcrb_aic3107S_ops,
 	},
 	{
 		.name = "SSP2 Capture Port",
@@ -245,19 +246,19 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.cpu_dai_name = "System Pin",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.init = NULL,
 		.dpcm_capture = 1,
 		.ignore_suspend = 1,
 		.nonatomic = 1,
 		.dynamic = 1,
-		.ops = &apli_lfcrb_aic3107S_ops,
+		.ops = &apli_lhcrb_aic3107S_ops,
 	},
 	{
 		.name = "SSP4 Playback Port",
 		.stream_name = "SSP4 Speaker",
 		.cpu_dai_name = "Deepbuffer Pin",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.nonatomic = 1,
 		.dynamic = 1,
 		.codec_name = "snd-soc-dummy",
@@ -265,7 +266,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.trigger = {SND_SOC_DPCM_TRIGGER_POST
 			, SND_SOC_DPCM_TRIGGER_POST},
 		.dpcm_playback = 1,
-		.ops = &apli_lfcrb_aic3107S_ops,
+		.ops = &apli_lhcrb_aic3107S_ops,
 	},
 	{
 		.name = "SSP4 Capture Port",
@@ -273,20 +274,20 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.cpu_dai_name = "Reference Pin",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.init = NULL,
 		.dpcm_capture = 1,
 		.ignore_suspend = 1,
 		.nonatomic = 1,
 		.dynamic = 1,
-		.ops = &apli_lfcrb_aic3107S_ops,
+		.ops = &apli_lhcrb_aic3107S_ops,
 	},
 	/* Back End DAI links */
 	{
 		/* SSP0 - Codec */
-		.name = "SSP0-Codec",
-		.be_id = 0,
-		.cpu_dai_name = "SSP0 Pin",
+		.name = "SSP1-Codec",
+		.id = 0,
+		.cpu_dai_name = "SSP1 Pin",
 		.codec_name = "tlv320aic3x-codec.3-0018",
 		.codec_dai_name = "tlv320aic3x-hifi",
 		.ignore_suspend = 1,
@@ -295,7 +296,7 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.dai_fmt = SND_SOC_DAIFMT_I2S |
 					SND_SOC_DAIFMT_NB_NF |
 					SND_SOC_DAIFMT_CBM_CFM,
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.ignore_suspend = 1,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
@@ -304,12 +305,12 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 	},
 	{
 		/* SSP1 - Codec */
-		.name = "SSP1-Codec",
-		.be_id = 1,
-		.cpu_dai_name = "SSP1 Pin",
+		.name = "SSP2-Codec",
+		.id = 1,
+		.cpu_dai_name = "SSP2 Pin",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.ignore_suspend = 1,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
@@ -318,15 +319,15 @@ static int aic3107_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 	},
 	{
 		/* SSP3 - Codec */
-		.name = "SSP3-Codec",
-		.be_id = 2,
-		.cpu_dai_name = "SSP3 Pin",
+		.name = "SSP4-Codec",
+		.id = 2,
+		.cpu_dai_name = "SSP4 Pin",
 		.codec_name = "snd-soc-dummy",
 		.codec_dai_name = "snd-soc-dummy-dai",
 		.dai_fmt = SND_SOC_DAIFMT_I2S |
 					SND_SOC_DAIFMT_NB_NF |
 					SND_SOC_DAIFMT_CBS_CFS,
-		.platform_name = "0000:00:0e.0",
+		.platform_name = pname,
 		.ignore_suspend = 1,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
@@ -394,7 +395,7 @@ static int apli_audio_remove(struct platform_device *pdev)
 	.probe = apli_audio_probe,
 	.remove = apli_audio_remove,
 	.driver = {
-		.name = "lfcrb_aic3107S_i2s",
+		.name = "lhcrb_aic3107S_i2s",
 	},
 };
 
@@ -403,4 +404,4 @@ static int apli_audio_remove(struct platform_device *pdev)
 /* Module information */
 MODULE_DESCRIPTION("Intel Audio aic3107 Machine driver for APL-I LH CRB");
 MODULE_LICENSE("GPL v2");
-MODULE_ALIAS("platform:lfcrb_aic3107S_i2s");
+MODULE_ALIAS("platform:lhcrb_aic3107S_i2s");
-- 
1.9.1

